main.py

```diff
@@ -0,0 +1,123 @@
+import os
+import urllib.parse
+import time
+import logging
+import re
+
+from github import Github
+from gitlab import Gitlab
+from prometheus_client import start_http_server
+from prometheus_client.core import REGISTRY
+from sentry_sdk import init as sentry_sdk_init
+
+from lib.saas_repo import get_saas_repos, SaasRepo
+from lib.repo import GHRepo, GLRepo, CommitNotFound
+from lib.metrics import SaasCollector
+from lib.gql import GqlApi
+
+logger = logging.getLogger()
+
+logging.basicConfig(format='%(levelname)s: %(message)s', level=logging.INFO)
+
+GITHUB_TOKEN = os.environ['GITHUB_TOKEN']
+GITLAB_SERVER = os.environ['GITLAB_SERVER']
+GITLAB_TOKEN = os.environ['GITLAB_TOKEN']
+
+APP_INTERFACE_URL = os.environ['APP_INTERFACE_URL']
+APP_INTERFACE_TOKEN = os.environ['APP_INTERFACE_TOKEN']
+
+SENTRY_DSN = os.getenv('SENTRY_DSN')
+
+gh_client = Github(GITHUB_TOKEN)
+gl_client = Gitlab(GITLAB_SERVER, private_token=GITLAB_TOKEN)
```

Comment:
this client initialization should probably move into the `lib.repo` module, inside the `__init__()` of the respective classes